File: qqrobot-spring-boot-starter/src/main/java/de/honoka/qqrobot/framework/impl/onebot/config/OnebotProperties.java
package de.honoka.qqrobot.framework.impl.onebot.config;

import de.honoka.sdk.util.file.FileUtils;
import lombok.Data;
import lombok.experimental.Accessors;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.nio.file.Paths;

@ConfigurationProperties("honoka.qqrobot.onebot")
@Data
@Accessors(chain = true)
public class OnebotProperties {
    
    private String host;
    
    private Integer httpPort;
    
    private Integer websocketPort;
    
    /**
     * 定义缓存文件所存放的目录。<br>
     *
     * 发送图片或文件前，需要先将InputStream中的数据写出到文件中，才能被OneBot服务使用。
     */
    private String cachePath = Paths.get(FileUtils.getMainClasspath(), "cache").normalize().toString();
}

File: qqrobot-spring-boot-starter/src/main/java/de/honoka/qqrobot/framework/impl/onebot/config/OnebotProperties.kt
package de.honoka.qqrobot.framework.impl.onebot.config

import kotlin.io.path.Path

//这些属性不能直接放在OnebotProperties类中作为字段，会被解析为Spring Boot配置项

val OnebotProperties.urlPrefix: String
    get() = "http://$host:$httpPort"

val OnebotProperties.websocketUrlPrefix: String
    get() = "ws://$host:$websocketPort"

val OnebotProperties.fileReceiverUrlPrefix: String
    get() = "http://$host:$fileReceiverPort"

val OnebotProperties.imagePath: String
    get() = Path(cachePath, "image").toString()

val OnebotProperties.fileToUploadPath: String
    get() = Path(cachePath, "fileToUpload").toString()